phpBB/assets/javascript/phpbb-avatars.js

```diff
@@ -11,11 +11,22 @@ phpbb.avatars = {
 	cropper: null,
 	image: null,
 
-	buttons: $('#avatar-cropper-buttons'),
-	box: $('#avatar-box'),
-	data: $('#avatar-cropper-data'),
-	input: $('#avatar_upload_file'),
-	driver: $('#avatar_driver'),
+	/** @type {jQuery} */
+	$buttons: $('#avatar-cropper-buttons'),
+
+	/** @type {jQuery} */
+	$box: $('#avatar-box'),
+
+	/** @type {jQuery} */
+	$data: $('#avatar-cropper-data'),
+
+	/** @type {jQuery} */
+	$input: $('#avatar_upload_file'),
+
+	/** @type {jQuery} */
+	$driver: $('#avatar_driver'),
+
+	/** @type {string} */
 	driverUpload: 'avatar_driver_upload',
```

Comment:
Is jshint driving this weird comment thing you are doing inline? Instead of in the jsdoc block? Also jshint needs removed as it's ancient and not well managed which is why I was working to replace it with xojs which is basically eslint in a better wrapper.

Reply:
We have a jsdoc block? This is in line with jsdoc btw, not based on jshint. If you think it makes more sense to sum this up into one block I'm open for suggestions, I did however like that this helps with static analysis in phpstorm.

Reply:
yes there is a jsdoc it works the same as all the others https://github.com/phpbb/chameleon/blob/master/all/js/temp.js

Reply:
Unfortunately, other elements don't seem to work properly for type hinting in phpstorm. Can you maybe propose something that would allow for proper type hinting?

Reply:
I think it's because you are building a single obj. al your params are essential just objs and your methods are just objs. I do not think you can jsdoc internal to an obj.